src/zarr/core/array.py

```diff
@@ -161,6 +161,77 @@ async def get_array_metadata(
 
 @dataclass(frozen=True)
 class AsyncArray:
+    """
+    An asynchronous array class representing a chunked array stored in a Zarr store.
+
+    Parameters
+    ----------
+    metadata : ArrayMetadata
+        The metadata of the array.
+    store_path : StorePath
+        The path to the Zarr store.
+    codec_pipeline : CodecPipeline, optional
+        The codec pipeline used for encoding and decoding chunks, by default None.
+    order : {'C', 'F'}, optional
+        The order of the array data in memory, by default None.
+
+    Attributes
+    ----------
+    metadata : ArrayMetadata
+        The metadata of the array.
+    store_path : StorePath
+        The path to the Zarr store.
+    codec_pipeline : CodecPipeline
+        The codec pipeline used for encoding and decoding chunks.
+    order : {'C', 'F'}
+        The order of the array data in memory.
+
+    Methods
+    -------
+    create(store, ...)
+        Create a new AsyncArray in the given store.
+    open(store, ...)
+        Open an existing AsyncArray from the given store.
+    resize(new_shape, ...)
+        Resize the AsyncArray to the specified shape.
+    update_attributes(new_attributes, ...)
+        Update the attributes of the AsyncArray.
+    info()
+        Print information about the AsyncArray.
+
+    Properties
```

Comment:
https://numpydoc.readthedocs.io/en/latest/format.html#class-docstring mentions that the docstrings for properties can go on the property, and the class just needs their name.

Reply:
thanks! I'd missed this